scrapy/Fetch/__init__.py

```diff
@@ -0,0 +1,46 @@
+import asyncio
+import logging
+import scrapy
+import warnings
+
+from twisted.internet.defer import Deferred
+
+from scrapy.core import config
+from scrapy.core import engine
+from scrapy.utils.log import logformatter_adapter, failure_to_exc_info
+from scrapy import signals
+from scrapy.core.downloader import Downloader 
+
+
+
+logger = logging.getLogger(__name__)
+
+
+
+class Fetch():
+    def __init__(self, Request):
+        self.Request = Request
+        self.crawler = engine.concraw
+        self.spider = engine.conspid
```

Comment:
There can be several Crawlers running at the same time, and several spiders; we can't use global variables. I think to access Crawler contextvars should be used.